src/Peachpie.Library/OpenSSL.cs

```diff
@@ -0,0 +1,741 @@
+
+            var aesAlg = PrepareCipher(ctx, dataBytes, key, cipher, iv, options);
+            ICryptoTransform encryptor = aesAlg.CreateEncryptor(aesAlg.Key, aesAlg.IV);
+
+            using MemoryStream msEncrypt = new MemoryStream();
+            using CryptoStream csEncrypt = new CryptoStream(msEncrypt, encryptor, CryptoStreamMode.Write);
+
+            using (StreamWriter swEncrypt = new StreamWriter(csEncrypt))
+                swEncrypt.Write(data);
+
+            encrypted = msEncrypt.ToArray();
+
+            if ((options & Options.OPENSSL_RAW_DATA) == Options.OPENSSL_RAW_DATA)
+                return new PhpString(encrypted);
+            else
+                return System.Convert.ToBase64String(encrypted);
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Gets the cipher initialization vector (iv) length.
+        /// </summary>
+        /// <param name="method">The cipher method, see openssl_get_cipher_methods() for a list of potential values.</param>
+        /// <returns>Returns the cipher length on success, or FALSE on failure.</returns>
+        [return: CastToFalse]
+        public static int openssl_cipher_iv_length(string method)
+        {
+            if (CiphersAliases.TryGetValue(method, out string methodName))
+                method = methodName;
+
+            Cipher cipherMethod;
+            if (!Ciphers.TryGetValue(method, out cipherMethod))
+            {
+                // Unknown cipher algorithm.
+                PhpException.Throw(PhpError.E_WARNING, Resources.LibResources.unknown_cipher);
+                return -1;
+            }
+
+            // In .NET there must be non zero length IV, but in PHP cipher mode ECB has zero length. 
+            // Due to compatibility in openssl_cipher_iv_length, there is this wierd thing.         
+            return (cipherMethod.Mode == CipherMode.ECB) ? 0 : cipherMethod.IVLength;
+        }
+
+        /// <summary>
+        /// Gets a list of available cipher methods.
+        /// </summary>
+        /// <param name="aliases">Set to TRUE if cipher aliases should be included within the returned array.</param>
+        /// <returns>An array of available cipher methods.</returns>
+        public static PhpArray openssl_get_cipher_methods(bool aliases = false)
+        {
+            PhpArray result = new PhpArray(Ciphers.Keys);
+
+            if (aliases)
+                result.AddRange(CiphersAliases);
+
+            return result;
+        }
+
+        #region openssl_digest/get_md_methods
+
+        // There are algos, which are implemented in .NET but there are not implemented in Hash.cs
+        private static Dictionary<string, Func<byte[], byte[]>> AditionaHashMethods = new Dictionary<string, Func<byte[], byte[]>>
+        {
+            {"sha384", (byte[] data) => SHA384Managed.Create().ComputeHash(data)}
+        };
+
+        private static Dictionary<string, string> HashAliases = new Dictionary<string, string>
```

Comment:
case sensitive?

Reply:
Yes, it is. I have already corrected it.